go/src/koding/kites/kloud/provider/provider.go

```diff
@@ -21,6 +21,10 @@ import (
 	"gopkg.in/mgo.v2/bson"
 )
 
+// All is a global lookup map used ny kloud to register
+// available stack providers.
+var All = make(map[string]func(*BaseProvider) stack.Provider)
```

Comment:
The thinking behind `provider.All` is it's meant to be modified in packages' init functions, so access is always synchronised. Sometimes (especially in tests, which I'm planning to add) I'd want to instrument existing providers - using functions instead of global map I'd need to also add `Get*` and `Delete*`.

I'm going to think about this more and eventually update in separate PR.